src/components/Table/AdvancedTable.tsx

```diff
@@ -1,33 +1,90 @@
 import styled from 'styled-components';
+import { useEffect, useState } from 'react';
+import { all } from 'api/firebase/firestore/firestore-actions';
+import { isEmpty } from 'others/helper-functions';
+import { BaseObject } from 'model/base-object';
+import { TableInstance, useFlexLayout, usePagination, useTable } from 'react-table';
+import { ColumnDefinition } from 'model/table/table-definitions';
 import TopBar from './TopBar';
-import Content, { ContentProps } from './Content';
+import Paginator from './Paginator';
+import Content from './Content';
 import Toolbar, { ToolbarProps } from './Toolbar';
 
 const Styles = styled.div`
   margin: 2rem;
 `;
 
-export interface AdvancedTableProps<T extends object> extends ToolbarProps, ContentProps<T> {
+export interface AdvancedTableProps<T extends BaseObject> extends ToolbarProps {
   name: string;
+  collection: string;
+  columns: ColumnDefinition<T>[];
+  filterColumns?: string[];
   showTopBar?: boolean;
   showToolbar?: boolean;
 }
 
-function AdvancedTable<T extends object>(props: AdvancedTableProps<T>) {
+function AdvancedTable<T extends BaseObject>(props: AdvancedTableProps<T>) {
+  const [data, setData] = useState<T[]>([]);
+  const [displayedData, setDisplayedData] = useState<T[]>([]);
+
+  useEffect(() => {
+    const docs = all<T>(props.collection);
+    docs
+      .then((categories) => {
+        setData(categories);
+        setDisplayedData(categories);
+      })
+      .catch((error) => {
+        throw new Error(`Could not fetch ${props.collection} collection!. Error: ${error.message}`);
+      });
+  }, [props.collection]);
+
+  const onSearch = (value: string): void => {
+    if (isEmpty(value) || isEmpty(props.filterColumns)) setDisplayedData(data);
+    else setDisplayedData(data.filter((row) => filterRow(row, value)));
+  };
+
+  const filterRow = (row: T, value: string): boolean => {
+    let show: boolean = false;
+    if (props.filterColumns !== undefined) {
+      for (let i = 0; i < props.filterColumns.length; i += 1) {
```

Comment:
Wouldn't it be better if you wrote that imperative code more declaratively, e.g. using the array filter method?

Reply:
We discussed at the meeting on 23.04.2021 that I'm leaving it this way for now.